code/modules/mob/living/carbon/xenomorph/castes/widow/abilities_widow.dm

```diff
@@ -0,0 +1,295 @@
+/datum/action/xeno_action/spider_swarm/action_activate()
+	. = ..()
+	if(!do_after(owner, 3 SECONDS, TRUE, owner, BUSY_ICON_DANGER))
+		return fail_activate()
+	/// We store the spiderling that we want to posses in order to make the other spiderlings follow it
+	current_controlling_spiderling = new(get_turf(owner), owner)
+	SEND_SIGNAL(owner, COMSIG_ESCORTED_ATOM_CHANGING, current_controlling_spiderling)
+	owner.mind.transfer_to(current_controlling_spiderling)
+	owner.doMove(null)
+	/// We want to access the spiderlings list and therefore have this
+	var/datum/action/xeno_action/create_spiderling/create_spiderling_action = owner.actions_by_path[/datum/action/xeno_action/create_spiderling]
+
+	for(var/spawned_spiderlings = 0, spawned_spiderlings < amount_of_spiderlings, spawned_spiderlings++)
+		var/mob/living/carbon/xenomorph/spiderling/new_spiderling = new /mob/living/carbon/xenomorph/spiderling(current_controlling_spiderling.loc, current_controlling_spiderling)
+		/// here we add the created spiderligns to the list
+		create_spiderling_action.add_spiderling(new_spiderling)
+
+	create_spiderling_action.add_spiderling(current_controlling_spiderling)
+	succeed_activate()
+	add_cooldown()
+
+/// This happens whenever the spiderling that we control dies and there are spiderlings left to control
+/datum/action/xeno_action/spider_swarm/proc/switch_to_next_spiderling(mob/living/carbon/xenomorph/spiderling/spiderling)
+	SEND_SIGNAL(current_controlling_spiderling, COMSIG_ESCORTED_ATOM_CHANGING, spiderling)
+	current_controlling_spiderling.mind.transfer_to(spiderling)
+	current_controlling_spiderling = spiderling
+
+/// Put the player back in widow
+/datum/action/xeno_action/spider_swarm/proc/switch_to_mother()
+	owner.forceMove(get_turf(current_controlling_spiderling))
+	current_controlling_spiderling.mind.transfer_to(owner)
+
+// ***************************************
+// *********** Burrow
+// ***************************************
+
+/datum/action/xeno_action/burrow
+	name = "Burrow"
+	ability_name = "Burrow"
+	mechanics_text = " Burrow into the ground to hide in plain sight "
+	action_icon_state = "burrow"
+	plasma_cost = 1
+	cooldown_timer = 1 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_BURROW
+/datum/action/xeno_action/burrow/action_activate()
+	. = ..()
+	var/mob/living/carbon/xenomorph/X = owner
+	if(X.widow_burrowed)
+		return
+
+	/// We need the list of spiderlings so that we can burrow them
+	var/datum/action/xeno_action/create_spiderling/create_spiderling_action = owner.actions_by_path[/datum/action/xeno_action/create_spiderling]
+	to_chat(X, span_xenowarning("We start burrowing into the ground"))
+	if(!do_after(X, 3 SECONDS, TRUE, target, BUSY_ICON_DANGER))
+		return fail_activate()
+	to_chat(X, span_xenowarning("We have burrowed ourselves, we are hidden from the enemy"))
+	/// This is the burrow code for Widow
+	X.alpha = 0
+	X.mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	X.density = FALSE
+	X.throwpass = TRUE
+	X.widow_burrowed = TRUE
```

Comment:
this should be a xenomorph proc. 

So you can use the same code for spiderlings/widow